### PR TITLE
Getstarted tweaks

### PR DESCRIPTION
**Pull request by londoncalling**
### Quick summary

@johndmulhausen I'm walking through the getting started, and here are edits so far, and expansion of some topics. The tutorial is beautifully written, and friendly. My goal is to keep with your style, but make some improvements where I had questions as a guinea pig user.

Adding @mstanleyjones  as a reviewer, for another viewpoint. (Still TBD for my walkthrough are parts 5 stacks, part 6 deploy.)

### What's changed

- copyedits
- added subtopics to longer steps like "Share your image", and showed more example outputs for commands
- more details and a screen snap of showing app in a web browser
- added output for `curl` command (so that this will be recognizable when they watch the asciinema replay of commands)
- added an example of image tagging syntax (the same one used on the asciicinema to better illustrate how repository and tag naming are used, and to map to the asciicinema replay of commands)
- added more example outputs and commands in general, where not having them could potentially confuse someone
- explained logging in to Docker and using Docker ID a little more
- improved the notes a bit, re-styled when they were breaking
- added more info on "cleanup" topics
- clarified Compose and Machine as prerequisites and how to get them
- @mstanleyjones review comments incorporated
- @johndmulhausen, updates per our discussion (reminders and code comments about replacing `username/repo:tag` with user-specific image details, added prerequisites to make sure docker machines are running and swarm is up, screen snaps of visualizer and web page with `redis` service, broke out steps on "Stacks" topic so that it's less likely users skim, and skip steps like creating a `data` directory)
- In part 6 under "create your swarm", fixed links to AWS, and added a link for Azure 

### Reviewers

@johndmulhausen @JimGalasyn @mstanleyjones 

@jkorab, added Docker Machine as a prerequisite on Linux systems in part 4 and subsequent parts, per discussion thread on #3449

### CODE REVIEW

**Comment by londoncalling** [comment]
@johndmulhausen @JimGalasyn Just checking to see if ya'll had time to review this, or if you don't want to merge it?

File: get-started/part4.md [hunk 1]
```diff
@@ -229,26 +241,50 @@ look:
...
+>
+> Keep in mind that in order to use the ingress network in the swarm,
+> you need to have the following ports open between the swarm nodes
+> before you enable swarm mode:
 >
 > - Port 7946 TCP/UDP for container network discovery.
 > - Port 4789 UDP for the container ingress network.
+{: .note-vanilla}
 
 ## Iterating and scaling your app
```
Comment (mstanleyjones) [inline get-started/part4.md]: More gerunds in the title

File: get-started/part4.md [hunk 2]
```diff
@@ -229,26 +241,50 @@ look:
...
+You can join any machine, physical or virtual, to this swarm, using the
+same `docker swarm join` command you used on `myvm2`, and capacity will be added
+to your cluster. Just run `docker stack deploy` afterwards, and your app will
+take advantage of the new resources.
+
+## Cleanup
+
+You can tear down the stack with `docker stack rm`. For example:
+
+```
```
Comment (mstanleyjones) [inline get-started/part4.md]: highlight hint

**Comment by londoncalling** [comment]
@mstanleyjones thanks I'll have a look at all these today. Right now I'm going to kick the Netlify preview. I don't know what the deal is with Netlify today.

File: get-started/index.md [hunk 3]
```diff
@@ -111,21 +111,21 @@ Consider this diagram comparing virtual machines to containers:
 
 ![Virtual machine stack example](https://www.docker.com/sites/default/files/VM%402x.png)
 
-Virtual machines run guest operating systems -- note the OS layer in each box.
-This is resource intensive, and the resulting disk image and application state is
-an entanglement of OS settings, system-installed dependencies, OS security
-patches, and other easy-to-lose, hard-to-replicate ephemera.
+Virtual machines run guest operating systems &#8212; note the OS layer in each
```
Comment (londoncalling) [inline get-started/index.md]: @mstanleyjones , yes those are emdashes to replace the `--` that were in the original

File: get-started/index.md [hunk 4]
```diff
@@ -111,21 +111,21 @@ Consider this diagram comparing virtual machines to containers:
...
-in a container image is the executable and its package dependencies, which
-never need to be installed on the host system. These processes run like native
+in a container image is the executable and its package dependencies, which never
+need to be installed on the host system. These processes run like native
 processes, and you can manage them individually by running commands like `docker
-ps` -- just like you would run `ps` on Linux to see active processes. Finally,
-because they contain all their dependencies, there is no configuration
+ps` &#8212; just like you would run `ps` on Linux to see active processes.
+Finally, because they contain all their dependencies, there is no configuration
 entanglement; a containerized app "runs anywhere."
```
Comment (londoncalling) [inline get-started/index.md]: @mstanleyjones once the app is deployed, it doesn't matter where it's running, right? The needed Linux components are containerized.

File: get-started/index.md [hunk 5]
```diff
@@ -149,22 +149,22 @@ To generate this message, Docker took the following steps:
 ...(snipped)...
 ```
 
-Now would also be a good time to make sure you are using version 1.13 or higher
+Now would also be a good time to make sure you are using version 1.13 or higher. Run `docker --version` to check it out.
```
Comment (londoncalling) [inline get-started/index.md]: Trying to keep the spirit of John's style, this sounds ok as is to me.

File: get-started/index.md (hunk 4 above)
Comment (mstanleyjones) [inline get-started/index.md]: Yeah I was just thinking that you can't run a Linux container on Windows Server using the Windows container kernel. I am sure you can just skip this feedback, I was just htinking "runs everywhere within reason". Also you can't run an armhf container on x86 architecture, etc. It's in the weeds.

File: get-started/part2.md [hunk 6]
```diff
@@ -179,10 +179,21 @@ docker run -p 4000:80 friendlyhello
 ```
 
 You should see a notice that Python is serving your app at `http://0.0.0.0:80`.
-But that message coming from inside the container, which doesn't know you mapped
-port 80 of that container to 4000, making the correct URL
-`http://localhost:4000`. Go there, and you'll see the "Hello World" text, the
-container ID, and the Redis error message.
+But that message is coming from inside the container, which doesn't know you
```
Comment (londoncalling) [inline get-started/part2.md]: I think it's okay. It is vivid.

File: get-started/part2.md [hunk 7]
```diff
@@ -218,53 +229,105 @@ docker stop 1fa4ab2cf395
...
-version. So, putting all that together, enter your username, and repo
-and tag names, so your existing image will upload to your desired destination:
+### Tag the image
+
+The notation for associating a local image with a repository on a registry is
+`username/repository:tag`. The tag is optional, but recommended, since it is
+the mechanism that registries use to give Docker images a version. Give the
+repository and tag meaningful names for the context, such as
+`get-started:part1`. This will put the image in the `get-started` repository and
+tag it as `part1`.
```
Comment (londoncalling) [inline get-started/part2.md]: I think it's correct, given this article: https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375

File: get-started/part2.md [hunk 8]
```diff
@@ -218,53 +229,105 @@ docker stop 1fa4ab2cf395
...
+3c1fe835fb6b: Already exists
+4c9f1fa8fcb8: Already exists
+ee7d8f576a14: Already exists
+fbccdcced46e: Already exists
+Digest: sha256:0601c866aab2adcc6498200efd0f754037e909e5fd42069adeff72d1e2439068
+Status: Downloaded newer image for john/get-started:part1
+ * Running on http://0.0.0.0:80/ (Press CTRL+C to quit)
+```
+
+>  **Note**: If you don't specify the `:tag` portion of these commands,
```
Comment (londoncalling) [inline get-started/part2.md]: I don't think it is indented ...

File: get-started/part4.md [hunk 9]
```diff
@@ -108,6 +109,17 @@ to open a terminal session on that VM. Type `exit` when you're ready to return
 to the host shell prompt. It may be easier to paste the join command in that
 way.
 
+You can `ssh` into your manager (`myvm1`), and type `docker node ls`:
```
Comment (londoncalling) [inline get-started/part4.md]: Well, as I was reading your comment, two developers were talking next to me and one literally said "somebody can ssh in" ..

Comment (mstanleyjones) [inline get-started/part4.md]: I know, it's commonly used. Just like "to grep" as a verb. But it's a nightmare for people who have a different native language than English, and it causes a lot of problems in translation.

**Comment by londoncalling** [comment]
@johndmulhausen  @JimGalasyn Hey y'all, I 'm going to go ahead and merge this, as it's been sitting for a while, and includes some fairly useful fixes. Feel free to log more updates on new PR's and reference things here as needed. Thanks @mstanleyjones for all the feedback.

**Pull request merged**